src/main/java/org/truffleruby/core/TruffleSystemNodes.java

```diff
@@ -273,4 +275,19 @@ protected int availableProcessors() {
 
     }
 
+    @CoreMethod(names = "allocated_bytes_of_current_thread", onSingleton = true)
+    public abstract static class AllocatedBytesNode extends CoreMethodArrayArgumentsNode {
+        private static ThreadMXBean bean;
+
+        @TruffleBoundary
+        @Specialization
+        protected static long allocatedBytes() {
+            if (bean == null) {
```

Comment:
I might comment here 'data races are ok - may duplicate an instance but that's not an issue'.

Reply:
Hmmm. I don't actually know what happens here in the JVM. I would expect to get the same bean instance every time, and `setThreadAllocatedMemoryEnabled(true)` to be idempotent.
Though, it seems somewhat platform specific.

Reply:
```
     * Returns the platform MXBean implementing
     * the given {@code mxbeanInterface} which is specified
     * to have one single instance in the Java virtual machine.
```
on `java.lang.management.ManagementFactory#getPlatformMXBean(java.lang.Class<T>)` used by `java.lang.management.ManagementFactory#getThreadMXBean`.
So `ManagementFactory.getThreadMXBean()` is a singleton.

(if it wasn't it could be an issue, as setThreadAllocatedMemoryEnabled() might not be called on all instances)